lib/puma/dsl.rb

```diff
@@ -46,6 +46,14 @@ def self.ssl_bind_str(host, port, opts)
         else ''
         end
 
+      if ['127.0.0.1', 'localhost'].include?(host)
+        key = "#{ENV["HOME"]}/.localhost/localhost.key"
+        cert = "#{ENV["HOME"]}/.localhost/localhost.crt"
```

Comment:
Just in case the paths ever change, I think we should do what we do in the Binder class does and ask the localhost gem for the paths (and only fallback to static paths if the gem can't answer that):

https://github.com/puma/puma/blob/af3675b2ee5485ec4fc62588869c41a5d5e278f7/lib/puma/binder.rb#L299-L309

(it does have fallback

Reply:
Would be good to not repeat the same code twice, but I'm not sure where would be a good place to share this code between Binder and DSL. @ye-lin-aung might have ideas?

Reply:
Agreed on both points

Reply:
Also the PR as-written I think does not respect the same conditions as the code in `Binder` - dev environment only, and localhost must be defined.

Reply:
Yeah, I was struggling with that a bit, because DSL is evaluated before rackup file is loaded. So, even if we have `require "localhost/authority"` in `.ru` file we can't check it on DSL because the file is not loaded yet. Any suggestions on how to tackle this problem?